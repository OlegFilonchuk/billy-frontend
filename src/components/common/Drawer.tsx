import React, { FC } from 'react';
import { TWTransitionDuration } from '../../types';
import { TRANSITION_DURATION } from '../../constats';
import TopPanelPlaceholder from './TopPanelPlaceholder';

type Props = {
  side: 'left' | 'right';
  open: boolean;
  transitionDuration?: TWTransitionDuration;
};

const Drawer: FC<Props> = ({
  side,
  open,
  children,
  transitionDuration = TRANSITION_DURATION,
}) => {
  const closedClassName =
    side === 'left' ? '-translate-x-full' : 'translate-x-full';

  return (
    <aside
      className={`transform top-0 ${
        side === 'left' ? 'left-0' : 'right-0'
      } w-52 bg-bg fixed h-full overflow-auto ease-in-out transition-all duration-${transitionDuration} z-40 ${
        open ? 'translate-x-0' : closedClassName
      }`}
    >
      <TopPanelPlaceholder />

      <div className="p-2">{children}</div>
    </aside>
  );
};

export default Drawer;
