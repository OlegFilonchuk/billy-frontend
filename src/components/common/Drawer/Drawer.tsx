import React, { FC } from 'react';
import { TWTransitionDuration } from '../../../types';
import { TRANSITION_DURATION } from '../../../constants';

export type DrawerProps = {
  side: 'left' | 'right';
  open: boolean;
  transitionDuration?: TWTransitionDuration;
};

const Drawer: FC<DrawerProps> = ({
  side,
  open,
  children,
  transitionDuration = TRANSITION_DURATION,
}) => {
  const closedClassName =
    side === 'left' ? '-translate-x-full' : 'translate-x-full';

  return (
    <aside
      className={`pt-16 transform top-0 ${
        side === 'left' ? 'left-0' : 'right-0'
      } w-52 bg-bg fixed h-full overflow-auto ease-in-out transition-all duration-${transitionDuration} z-40 ${
        open ? 'translate-x-0' : closedClassName
      }`}
    >
      <div className="p-4">{children}</div>
    </aside>
  );
};

export default Drawer;
