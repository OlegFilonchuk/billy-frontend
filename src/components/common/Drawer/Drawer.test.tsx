import React from 'react';
import { RenderResult, render } from '@testing-library/react';
import Drawer from './Drawer';

describe('Drawer', () => {
  const side = 'left';
  let ui: RenderResult;
  let open: boolean;

  describe('in the open state', () => {
    beforeEach(() => {
      open = true;
      ui = render(<Drawer side={side} open={open} />);
    });

    it('should render properly', () => {
      expect(ui.container).toMatchSnapshot();
    });
  });

  describe('in the closed state', () => {
    beforeEach(() => {
      open = false;
      ui = render(<Drawer side={side} open={open} />);
    });

    it('should render properly', () => {
      expect(ui.container).toMatchSnapshot();
    });
  });
});
