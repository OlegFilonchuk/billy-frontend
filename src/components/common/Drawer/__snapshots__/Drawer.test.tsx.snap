// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer in the closed state should render properly 1`] = `
<div>
  <aside
    class="pt-16 transform top-0 left-0 w-52 bg-bg fixed h-full overflow-auto ease-in-out transition-all duration-300 z-40 -translate-x-full"
  >
    <div
      class="p-4"
    />
  </aside>
</div>
`;

exports[`Drawer in the open state should render properly 1`] = `
<div>
  <aside
    class="pt-16 transform top-0 left-0 w-52 bg-bg fixed h-full overflow-auto ease-in-out transition-all duration-300 z-40 translate-x-0"
  >
    <div
      class="p-4"
    />
  </aside>
</div>
`;
